### DISCUSSION TITLE
Any plans to write Lobster in Lobster?

### DISCUSSION THREAD

**Discussion by Tembocs** [aardappel/lobster#163] [General]
Are there any plans to write Lobster in Lobster?

**Comment by aardappel** [comment]
Definitely thought about it, it's just a ton of work. Could be done incrementally, first the parser, etc. The type checker in particular is complicated and would take a bit of work to translate.

On the positive side, Lobster is quite wonderful for language implementation, so should be fun.

**Comment by Xrayez** [comment]
What's the benefit of doing so? I'm legit curious.
* Faster development of Lobster itself and easier modification?
* Improve reflection capabilities of the language?
* Creating entire new languages via Lobster?

Or all combinations of the above? 🙂

**Reply by aardappel** [reply]
Just the first item. I am a LOT quicker writing code in Lobster than C++, and it would likely reduce bugs too.

Still, don't see this happening any time soon unless I magically get a sea of time to dedicate to the language.
